xbmc/cores/VideoPlayer/DVDInputStreams/DVDFactoryInputStream.cpp

```diff
@@ -153,6 +153,9 @@ std::shared_ptr<CDVDInputStream> CDVDFactoryInputStream::CreateInputStream(IVide
     if (finalFileitem.IsType(".m3u8"))
       return std::shared_ptr<CDVDInputStreamFFmpeg>(new CDVDInputStreamFFmpeg(finalFileitem));
 
+    if (finalFileitem.IsPVR() && fileitem.GetProperty("inputstreamclass").asString() == "inputstream.ffmpeg")
```

Comment:
> The `ColumnLimit` in `.clang-format` is set to `100` which defines line length [...]

https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#3-formatting

Reply:
this is not a pvr thingy -> test for pvr is wrong.
explicit requests for classes need to be done right after addons are checked -> move this up

Reply:
"inputstream.ffmpeg" this should be documented in some api file. how should an addon dev know how internal classes are named?

Reply:
check out cores/VideoPlayer/Interface/Addon/

Reply:
Sorry, meant to remove PVR check. If I move it up with misses the Curl check to set the DynPath.

Reply:
> If I move it up it misses the Curl check to set the DynPath.

fileitems are passed to inputstream, that includes dynPath. that also works for other inputstream classes like addons.

Reply:
Ok, works fine for internet based m3u8 files. If I copy one local I get:
`ffmpeg[0x7fd13dc20a40X]: [https] Protocol 'https' not on whitelist 'file,crypto'!`

Reply:
It feels wrong to limit `inputstreamclass` to internet streams only.

Reply:
Ok, added  `av_dict_set(&options, "protocol_whitelist", "file,http,https,tcp,tls,crypto", 0);` as the whitelist in `GetFFMpegOptionsFromInput` and it works with local m3u8 files. 

However I don't know if this is acceptable or not.